sqlx-mysql/src/types/json.rs

```diff
@@ -47,14 +47,14 @@ where
         buf.extend_from_slice(&[0u8; 9]);
 
         let encode_start = buf.len();
-        self.encode_to(buf);
+        self.encode_to(buf)?;
```

Comment:
This would leave the arguments buffer in an invalid state on returning an error, which could cause a problem if someone tries to reuse it or retry the encoding for some reason.

It should be sure to truncate `buf` to `lenenc_start` on error.

Reply:
Alternatively, the implementation of `MySqlArguments::add()` needs to defend against this, perhaps using a RAII guard.

Reply:
Yeah, agreed, this should be a more general solution, not something that needs to be taken into account for every single encode-Implementation.

Reply:
@FSMaxB were you going to leave this like it is?

Reply:
Nvm I see it's handled in `PgArguments:;add()`.